src/pages/write-for-egghead/topics.tsx

```diff
@@ -0,0 +1,175 @@
+                - [https://kentcdodds.com/blog/prop-drilling](https://kentcdodds.com/blog/prop-drilling)
+                - Use [React's Context API](https://kentcdodds.com/blog/how-to-use-react-context-effectively) for things that are truly necessary deep in the react tree.
+                - Redux took off, is because they solved this problem. So context is this problem solved and blessed by the framework.
+                - You can take this state that needs to be shared across multiple elements in my React tree and stick it into some registry that Reacts, maintains, and then anywhere from that part of the tree on down, will have access to that state.`,
+      },
+      {
+        headline:
+          'How do you decide when to use useEffect and when to use useState?',
+        points: `- useEffect is most commonly used with props when something is changing that you need to react to in a way other than displaying data on the screen.
+                - For example, if a prop comes in that has the component re-fetch something from the server and show a loading spinner while you wait for the response. This is more than just a display change, so it calls for useEffect.`,
+      },
+    ],
+  },
+  {
+    topicTitle: 'Advanced React',
+    topics: [
+      {
+        headline: 'Where should State live?',
+        points: `- You should never have State in two places at any time. Do not duplicate State.
+                - You need to have a single source of truth, and it needs to live in one place.
+                    - It can live in memory, local storage, session storage, URL bar and the browser history.
+                    - Example of implementing State in browser history.`,
+      },
+      {
+        headline: 'How do determine the right amount of UI abstraction?',
+        points: `- For client state: there's no need to go outside of React. Unless I want to, like going for Redux because of the dev tools.
+                - For server-state: I'm going to reach for React Query, Apollo, SWR.
+                - Keeping track of local state is easy, it's predictable. You can control everything inside your application.
+                - When you start consuming stuff outside of your app, you are no longer in control. They are reading a snapchat of an API endpoint. People treat data as they own it, which they don't. And that's really hard to keep track of.
+                - Rule: Who owns this code?
+                    - If you can answer this question correctly, then you'll know what tool to use.`,
+      },
+      {
+        headline: 'When is XState not appropriate?',
+        points: `- You don't need a library to implement State Machines.
+                - You can use patters in each language.
+                - XState becomes more useful when representing Statecharts.`,
+      },
+    ],
+  },
+]
+
+export function TopicListComponent() {
```

Comment:
```suggestion
export default function TopicListComponent() {
```

You can run `yarn build` if you ever want to see output about what the error is